src/goto-symex/renaming_level.cpp

```diff
@@ -46,8 +46,12 @@ operator()(ssa_exprt ssa_expr, const namespacet &ns, unsigned thread_nr) const
 renamedt<ssa_exprt, L1> symex_level1t::
 operator()(renamedt<ssa_exprt, L0> l0_expr) const
 {
-  if(!l0_expr.get().get_level_1().empty())
+  if(
+    !l0_expr.get().get_level_1().empty() ||
+    !l0_expr.get().get_level_2().empty())
```

Comment:
To make sure there is no change in behaviour this will require an `INVARIANT` that `!l0_expr.get().get_level_2().empty()` implies `!l0_expr.get().get_level_1().empty()`. And I'm not convinced that holds, because a global variable would not need L1 renaming, even when L2 is non-empty.

Reply:
Yes this does not hold, so we can't add an invariant. The reason this does not change the behaviour is that it is only called from `set_l2_indices` which had the same check until this commit. (Note that we introduced the `symex_level2t::operator()` only 3 days ago).